src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Analyzer.cs

```diff
@@ -0,0 +1,205 @@
+            {
+                return;
+            }
+
+            IMethodSymbol? startsWithStringMethod = stringType.GetMembers(StringStartsWithMethodName).OfType<IMethodSymbol>().GetFirstOrDefaultMemberWithParameterInfos(stringParameter);
+            if (startsWithStringMethod == null)
+            {
+                return;
+            }
+
+            IMethodSymbol? compareToStringMethod = stringType.GetMembers(StringCompareToMethodName).OfType<IMethodSymbol>().GetFirstOrDefaultMemberWithParameterInfos(stringParameter);
+            if (compareToStringMethod == null)
+            {
+                return;
+            }
+
+            // Retrieve the StringComparer properties that need to be flagged: CurrentCultureIgnoreCase, InvariantCultureIgnoreCase
+
+            IEnumerable<IPropertySymbol> ccicPropertyGroup = stringComparerType.GetMembers(StringComparisonCurrentCultureIgnoreCaseName).OfType<IPropertySymbol>();
+            if (!ccicPropertyGroup.Any())
+            {
+                return;
+            }
+
+            IEnumerable<IPropertySymbol> icicPropertyGroup = stringComparerType.GetMembers(StringComparisonInvariantCultureIgnoreCaseName).OfType<IPropertySymbol>();
+            if (!icicPropertyGroup.Any())
+            {
+                return;
+            }
+
+            context.RegisterOperationAction(context =>
+            {
+                IInvocationOperation caseChangingInvocation = (IInvocationOperation)context.Operation;
+                IMethodSymbol caseChangingMethod = caseChangingInvocation.TargetMethod;
+
+                if (!caseChangingMethod.Equals(toLowerParameterlessMethod) &&
+                    !caseChangingMethod.Equals(toLowerInvariantParameterlessMethod) &&
+                    !caseChangingMethod.Equals(toUpperParameterlessMethod) &&
+                    !caseChangingMethod.Equals(toUpperInvariantParameterlessMethod))
+                {
+                    return;
+                }
+
+                if (caseChangingInvocation.Parent is not IInvocationOperation diagnosableInvocation)
+                {
+                    return;
+                }
+
+                var diagnosableMethod = diagnosableInvocation.TargetMethod;
+
+                DiagnosticDescriptor rule;
+                if (diagnosableMethod.Equals(containsStringMethod) ||
+                    diagnosableMethod.Equals(indexOfStringMethod) ||
+                    diagnosableMethod.Equals(startsWithStringMethod))
+                {
+                    rule = RecommendCaseInsensitiveStringComparisonRule;
+                }
+                else if (diagnosableMethod.Equals(compareToStringMethod))
+                {
+                    rule = RecommendCaseInsensitiveStringComparerRule;
+                }
+                else
+                {
+                    return;
+                }
+
+                context.ReportDiagnostic(diagnosableInvocation.CreateDiagnostic(rule, diagnosableMethod.Name));
```

Comment:
One of the two rules have argument, and the other doesn't. I personally wouldn't pass the argument in both cases.

Reply:
Good catch. I fixed this by moving the diagnostic creation to the above if else.